recipes/flex/all/conanfile.py

```diff
@@ -67,29 +68,37 @@ def _configure_autotools(self):
             "ac_cv_func_malloc_0_nonnull=yes", "ac_cv_func_realloc_0_nonnull=yes",
             # https://github.com/easybuilders/easybuild-easyconfigs/pull/5792
             "ac_cv_func_reallocarray=no",
-        ]
-
-        autotools.configure(args=configure_args, configure_dir=self._source_subfolder)
-        return autotools
+        ])
+        at.generate()
 
     def build(self):
-        autotools = self._configure_autotools()
+        apply_conandata_patches(self)
+        autotools = Autotools(self)
+        autotools.configure()
         autotools.make()
 
     def package(self):
-        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
-        autotools = self._configure_autotools()
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        autotools = Autotools(self)
         autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
 
     def package_info(self):
         self.cpp_info.libs = ["fl"]
+        self.cpp_info.set_property("cmake_file_name", "FLEX")
+        self.cpp_info.set_property("cmake_target_name", "FLEX::FLEX")
+        self.cpp_info.set_property("pkg_config_name", "flex")
+
+        self.cpp_info.names["cmake_find_package"] = "FLEX"
+        self.cpp_info.names["cmake_find_package_multi"] = "FLEX"
 
         bindir = os.path.join(self.package_folder, "bin")
         self.output.info("Appending PATH environment variable: {}".format(bindir))
         self.env_info.PATH.append(bindir)
+        self.buildenv_info.prepend_path("PATH", bindir)
```

Comment:
I think this line is not needed because it's done implicitly already - `env_info.PATH` is needed for legacy downstream consumers, but downstream consumers should already have the bindirs of this dependency in their `VirtualBuildEnv`s when consumed as a `tool_requires`,